lib/engine/src/serialize.rs

```diff
@@ -1,105 +0,0 @@
-use loupe::MemoryUsage;
-use serde::de::{Deserializer, Visitor};
-use serde::ser::Serializer;
```

Comment:
`serde` may still be good to keep around because it's very generic and widely used

Reply:
Our serialization format should be universal. That means that if someone serializes with Wasmer as a library, the result when reading from anywhere else should succeed.
Because of that, the serialization mechanism should be obscure to our users outside Wasmer.